### PR TITLE
Signing:  handle exception under Mono

### PR DESCRIPTION
**Pull request by dtivel**
Fix https://github.com/NuGet/Home/issues/6634.

### CODE REVIEW

**Review by PatoBeltran — APPROVED** [review]

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs [hunk 1]
```diff
@@ -78,5 +75,54 @@ public void DotnetRestore_SolutionRestoreVerifySolutionDirPassedToProjects()
...
+                        "netstandard1.3",
+                        new Dictionary<string, string>(),
+                        attributes);
+
+                    ProjectFileUtils.WriteXmlToFile(xml, stream);
+                }
+
+                var args = $"--source \"{packageSourceDirectory.Path}\" ";
+
+                _msbuildFixture.RestoreProject(workingDirectory, projectName, args);
```
Comment (rohit21agrawal) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs:117]: this test is not asserting anything.

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs [hunk 2]
```diff
@@ -78,5 +75,54 @@ public void DotnetRestore_SolutionRestoreVerifySolutionDirPassedToProjects()
...
+            }
+        }
+
+        private static byte[] GetResource(string name)
```
Comment (rohit21agrawal) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs:121]: 🏅

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs [hunk 3]
```diff
@@ -2220,10 +2220,88 @@ var targetFileContents
...
+  <PropertyGroup Condition="" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' "">
+    <OutputPath>bin\Debug\</OutputPath>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include=""System"" />
+    <Reference Include=""System.Core"" />
+    <Reference Include=""Microsoft.CSharp"" />
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include=""TestPackage.AuthorSigned"">
```
Comment (rohit21agrawal) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs:2267]: what is the source of this package?

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs [hunk 4]
```diff
@@ -2220,10 +2220,88 @@ var targetFileContents
...
                 // Assert
-                Assert.True(0 == r.Item1, r.Item2 + " " + r.Item3);               
+                Assert.True(0 == r.Item1, r.Item2 + " " + r.Item3);
                 var packageFileA = Path.Combine(randomSolutionFolder, "GlobalPackages", "packagea","1.1.0", "packageA.1.1.0.nupkg");
                 Assert.True(File.Exists(packageFileA));
             }
         }
+
+        [Fact]
+        public void RestoreCommand_WithAuthorSignedPackage_Succeeds()
```
Comment (jainaashish) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs:2233]: Wouldn't this test run on Mono?

**Review by jainaashish — APPROVED** [review]

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs (hunk 1 above)
Comment (nkolev92) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs:117]: The MsbuildIntegrationTestFixture asserts the exit code inside, doesn't it? :)
You should know 😃

**Review by nkolev92 — APPROVED** [review]

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs (hunk 1 above)
Comment (dtivel) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs:117]: As @nkolev92 said, `MsbuildIntegrationTestFixture .RestoreProject(...)` [asserts](https://github.com/NuGet/NuGet.Client/blob/023fe7670796a8986bbfdc520029e4cf0a6bbfda/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuilldIntegrationTestFixture.cs#L143) restore success.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs (hunk 3 above)
Comment (dtivel) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs:2267]: It's a static resource [included in this PR](https://github.com/NuGet/NuGet.Client/pull/2074/files#diff-d708b4762ffd5d9e1e514cffb32bf6d2).

This test first [extracts the static resource to disk](https://github.com/NuGet/NuGet.Client/pull/2074/files#diff-efde37b61c6a1135b7b8f64f080c4d28R2239), then [passes the directory as a package source](https://github.com/NuGet/NuGet.Client/pull/2074/files#diff-efde37b61c6a1135b7b8f64f080c4d28R2280).

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs (hunk 4 above)
Comment (dtivel) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs:2233]: Yes.  The CI test run shows that this test passed on Mono.

The expectation at this point is that signed package validation will fail but the package will still restore as an unsigned package, at least until https://github.com/mono/mono/issues/7456 is fixed.

**Pull request merged**